docs/user-guide/setting-up-ci-cd.rst

```diff
@@ -0,0 +1,207 @@
+:desc: Set up a CI/CD pipeline to ensure that iterative improvements are tested and deployed with minimum manual effort
+
+.. _setting-up-ci-cd:
+
+Setting up CI/CD
+================
+
+Developing a bot looks a little different than developing a traditional
+software application, but that doesn't mean you should abandon software
+development best practices. Setting up CI/CD ensures that incremental updates
+to your bot are improving it, not harming it.
+
+.. contents::
+   :local:
+   :depth: 2
+
+
+Overview
+--------
+
+Continous Integration (CI) is the practice of merging in code changes
+frequently, and automatically testing changes as they are committed. Continuous
+Deployment (CD) means automatically deploying integrated changes to a staging
+or production environment. Together, they allow for more frequent improvements
+to your assistant with less manual effort expended on testing and deployment.
+
+This guide will cover **what** should go in a CI/CD pipeline, specific to a
+Rasa project. **How** you implement that pipeline is up to you. There are many
+CI/CD tools out there, and you might already have a preferred one. Many Git
+repository hosting services like Github, Gitlab, and Bitbucket also provide
+their own CI/CD tools that you can make use of. 
+
+Continuous Integration
+----------------------
+
+Assistants are best improved with frequent, `incremental updates
+<https://rasa.com/docs/rasa-x/user-guide/improve-assistant/>`_.
+No matter how small a change is, you want to be sure that it doesn't introduce
+new problems or negatively impact the performance of your model. Most tests are
+quick enough to run on every change. However, you can set some more
+resource-intentsive tests to run only when certain files have been changed or
+when a certain tag is present.
+
+CI checks usually run on commit, or on merge/pull request.
+
+.. contents::
+   :local:
```

Comment:
We should make sure all of the titles are structured the same, 
![image](https://user-images.githubusercontent.com/22333512/78588207-81404600-783e-11ea-9d82-3999f9d5f704.png)

right now we have active verbs, imperative verbs and nouns. I'd lean toward imperative for a pipeline as you're telling it what to do, i.e.

Validate Data
Validate Stories
Train a Model
Test the Assistant
Compare NLU Performance
Test Action Code